include/seqan3/alphabet/structure/rna_structure_concept.hpp

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2006-2017, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2017, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup structure
+ * \author Joerg Winkler <j.winkler AT fu-berlin.de>
+ * \brief Provides seqan3::rna_structure_concept.
+ */
+
+#pragma once
+
+#include <seqan3/alphabet/concept.hpp>
+
+// ============================================================================
+// concept
+// ============================================================================
+
+namespace seqan3
+{
+
+/*!\brief A concept that indicates whether an alphabet represents RNA structure.
+ *!\ingroup structure
+ *!\details RNA structure alphabets are required to represent interactions among RNA nucleotides.
+ * Therefore, each structure letter can be categorised as unpaired, opening an interaction, or closing an interaction.
+ * Additionally, the ability of representing pseudoknots is a property of RNA structure types.
+ */
+template <typename type>
+concept bool rna_structure_concept = requires(type val)
+{
+    //! requires fulfillment of alphabet concept
+    requires alphabet_concept<type>;
+
+    { val.is_pair_open() };
+    { val.is_pair_close() };
+    { val.is_unpaired() };
+    type::pseudoknot_support;
```

Comment:
Similar to the other concepts these should be free functions/metafunctions. You can still implement them as member functions, but then add a free function → member function wrapper in `alphabet/detail/member_exposure.hpp`

Reply:
Okay, I implemented this. I still do not understand, why these very specific functions should spam the global function namespace? Also, for the rna_structure_concept it is confusing that it requires free functions, although all alphabets implement the member functions (so I put a comment that tells to look into `member_exposure.hpp`).